File: Backend/src/Modules/ReservationModule/Shared/Dtos/ReservationDto.cs
using System.Text.Json.Serialization;
using src.Modules.ReservationModule.Domain.Entities;

namespace src.Modules.ReservationModule.Shared.Dtos;

public class ReservationDto
{
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public string? Id { get; init; }
    public required string RoomId { get; init; } 
    public required string ReservationType { get; init; }
    public required TimeSlotDto TimeSlotDto { get; init; }
    public required DateTime Day { get; init; }
    
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public string? DeviceId { get; init; }
    
    // TODO reserved for event id
    // [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    // public string? EventId { get; init; }
}

File: Frontend/web/src/models/reservationDto.ts
import {TimeSlotDto} from "./timeSlotDto.ts";

export interface ReservationDto {
    id: string | null;
    roomId: string;
    reservationType: string;
    timeSlotDto: TimeSlotDto;
    day: Date;
    deviceId: string | null; // Optional, equivalent to `[JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]`
}